# PyCitySchools

## <span style="color:green">By Rodrigo Guarneros</span>

  * Here you can see the trends I found during the analysis
  
  * The trends are featured in text boxes like this with the title in green color  
#### <span style="color:green">Welcome!</span>

In [4]:
import pandas as pd

In [5]:
schools_fw_path = "Resources/schools_complete.csv"
students_fw_path = "Resources/students_complete.csv"

In [7]:
schools_fw = pd.read_csv(schools_fw_path)
students_fw = pd.read_csv(students_fw_path)

In [8]:
schools_fw.head()

School ID            school_name      type  size   budget
0          0      Huang High School  District  2917  1910635
1          1   Figueroa High School  District  2949  1884411
2          2    Shelton High School   Charter  1761  1056600
3          3  Hernandez High School  District  4635  3022020
4          4    Griffin High School   Charter  1468   917500

In [9]:
students_fw.head()

Student ID       student_name gender grade        school_name  \
0           0       Paul Bradley      M   9th  Huang High School   
1           1       Victor Smith      M  12th  Huang High School   
2           2    Kevin Rodriguez      M  12th  Huang High School   
3           3  Dr. Richard Scott      M  12th  Huang High School   
4           4         Bonnie Ray      F   9th  Huang High School   

   reading_score  math_score  
0             66          79  
1             94          61  
2             90          60  
3             67          58  
4             97          84

In [274]:
# 1. number of schools 15
schools = len(schools_fw["school_name"].unique())
print(schools)

15


In [306]:
# 2. Number of tudents: 39170
students = len(students_fw["student_name"])
print(students)

39170


In [74]:
# 3. Total Budget: 24649428
Total_Budget = schools_fw["budget"].sum()
print(Total_Budget)

24649428


In [20]:
schools_fw.index

RangeIndex(start=0, stop=15, step=1)

In [96]:
# 4. Calculate the average math score 
Average_Math = round(students_fw["math_score"].mean(),2)
print(Average_Math)

78.99


In [278]:
# 5. Calculate the average reading score
Average_Reading = round(students_fw["reading_score"].mean(),2)
print(Average_Reading)

81.88


In [277]:
# 6. Calculate the percentage of students with a passing math score (70 or greater)
pass_students_math = students_fw["math_score"]>=70

students_fw_pass = students_fw[pass_students_math] 

percent_pass_math = round(int(len(students_fw_pass))/int(len(students_fw))*100,2)

Percent_Pass_Math = print(f"{percent_pass_math}%")

74.98%


In [307]:
# 7. Calculate the percentage of students with a passing reading score (70 or greater)
pass_students_reading = students_fw["reading_score"]>=70

students_fw_pass_r = students_fw[pass_students_reading] 

percent_pass_read = round(int(len(students_fw_pass_r))/int(len(students_fw))*100,2)

Percent_Pass_Read = print(f"{percent_pass_read}%")


85.81%


In [87]:
# 8. Calculate the percentage of students who passed math and reading (% Overall Passing)

read_write = pass_students_math & pass_students_reading

students_2pass = students_fw[read_write]

percent_rw = round(int(len(students_2pass))/int(len(students_fw))*100,2)

Percent_Pass = print(f"{percent_rw}%")


65.17%


In [141]:
# 9. Create a dataframe to hold the above results
summary_list = {"Name":["Total Schools", "Total Students", "Total Budget", "Average Math Score", "Average Reading Score", "% Passing Math", "% Passing Reading", "% Overall Pasing"],
                "Value":[schools,students, Total_Budget, Average_Math, Average_Reading, f"{percent_pass_math}%", f"{percent_pass_read}%", f"{percent_rw}%"]}
df_summary = pd.DataFrame(summary_list)
df_summary

Name     Value
0          Total Schools        15
1         Total Students     39170
2           Total Budget  24649428
3     Average Math Score     78.99
4  Average Reading Score     81.88
5         % Passing Math    74.98%
6      % Passing Reading    85.81%
7       % Overall Pasing    65.17%

# <span style="color:green">General View</span>

  * At first glance, the general information show us 15 schools with a total budget of $24,649,428 and 39,170 students with several values about average math and reading score, as well as the share of students with a pass score in math (74.98%) and reading (85.81%), as we would expected. This performance of the students is more worrisome if we obtain both scores for each student and calculate the percentage of students with a passing score in both subjects (reading and mathematics), wich tell us that only 6.5 in every 10 students are successful in both subjets.

* Having in mind this general backgroun. Can we see in data more information about the performance in the schools and intudents? Budget or characteristics of the school are drivers about the student's performance?

In [130]:
df_summary.shape

(8, 2)

In [1]:
# Dependencies and Setup
import pandas as pd

# File to Load (Remember to Change These)
school_data_to_load = "Resources/schools_complete.csv"
student_data_to_load = "Resources/students_complete.csv"

# Read School and Student Data File and store into Pandas DataFrames
school_data = pd.read_csv(school_data_to_load)
student_data = pd.read_csv(student_data_to_load)

# Combine the data into a single dataset.  
school_data_complete = pd.merge(student_data, school_data, how="left", on=["school_name", "school_name"])

## District Summary

* Calculate the total number of schools

* Calculate the total number of students

* Calculate the total budget

* Calculate the average math score 

* Calculate the average reading score

* Calculate the percentage of students with a passing math score (70 or greater)

* Calculate the percentage of students with a passing reading score (70 or greater)

* Calculate the percentage of students who passed math **and** reading (% Overall Passing)

* Create a dataframe to hold the above results

* Optional: give the displayed data cleaner formatting

In [156]:
# Alternative
summary_dict = {"Total Schools":schools, "Total Students":students, "Total Budget":Total_Budget, "Average Math Score":Average_Math, "Average Reading Score": Average_Reading, 
                "% Passing Math":percent_pass_math, "%Passing Reading":percent_pass_read, "% Overall Pasing":percent_rw}
df_summary2 = pd.DataFrame.from_dict(summary_dict, orient="Index").T
df_summary2

Total Schools  Total Students  Total Budget  Average Math Score  \
0           15.0         39170.0    24649428.0               78.99   

   Average Reading Score  % Passing Math  %Passing Reading  % Overall Pasing  
0                  81.88           74.98             85.81             65.17

## School Summary

* Create an overview table that summarizes key metrics about each school, including:
  * School Name
  * School Type
  * Total Students
  * Total School Budget
  * Per Student Budget
  * Average Math Score
  * Average Reading Score
  * % Passing Math
  * % Passing Reading
  * % Overall Passing (The percentage of students that passed math **and** reading.)
  
* Create a dataframe to hold the above results

In [164]:
schools_fw.columns

Index(['School ID', 'school_name', 'type', 'size', 'budget'], dtype='object')

In [221]:
students_fw.shape

(39170, 7)

In [283]:
merge = students_fw.merge(schools_fw, how="left")
merge.columns

Index(['Student ID', 'student_name', 'gender', 'grade', 'school_name',
       'reading_score', 'math_score', 'School ID', 'type', 'size', 'budget'],
      dtype='object')

In [493]:
#•Create an overview table that summarizes key metrics about each school, including: 
schools_summary = merge.groupby(merge["school_name"]).mean() 
every_obs_is_there = students_fw[~merge["Student ID"].isin(students_fw["Student ID"])] 
#◾School Name 
School_Name = schools_fw.set_index("school_name").groupby(["school_name"]) 
#◾School Type 
School_Type = schools_fw.set_index("school_name")["type"] 
#◾Total Students 
Total_students = merge.set_index("school_name").groupby(["school_name"])["Student ID"].count()
Total_students = pd.DataFrame(Total_students)
Total_students = Total_students.rename(columns={"Student ID":"Total Students"})
#◾Total School Budget 
Total_budget_school = schools_fw.set_index("school_name")["budget"] 
Total_budget_school = pd.DataFrame(Total_budget_school)
Total_budget_school = Total_budget_school.rename(columns={"budget":"Total Budget School"})

#◾Per Student Budget 
Budget_pstudent = Total_budget_school["Total Budget School"]/Total_students["Total Students"]
Budget_pstudent = pd.DataFrame(Budget_pstudent)
Budget_pstudent = Budget_pstudent.rename(columns={0:"Per Student Budget"})
#◾Average Math Score 
Average_math_school = merge.set_index("school_name").groupby(["school_name"])["math_score"].mean() 
Average_math_school = pd.DataFrame(Average_math_school)
Average_math_school = Average_math_school.rename(columns={"math_score":"Average Math Score"})
#◾Average Reading Score 
Average_reading_school = merge.set_index("school_name").groupby(["school_name"])["reading_score"].mean() 
Average_reading_school = pd.DataFrame(Average_reading_school)
Average_reading_school = Average_reading_school.rename(columns={"reading_score":"Average Reading Score"})
#◾% Passing Math 
number_pass_math = merge.loc[merge["math_score"]>=70].groupby(["school_name"])["math_score"].count()
number_pass_math = pd.DataFrame(number_pass_math)
number_pass_math = number_pass_math.rename(columns={"math_score":"Passing Math"})
percent_pass_math = (number_pass_math["Passing Math"]/Total_students["Total Students"])*100
percent_pass_math = pd.DataFrame(percent_pass_math)
percent_pass_math = percent_pass_math.rename(columns={0:"% Passing Math"})
#◾% Passing Reading 
number_pass_reading = merge.loc[merge["reading_score"]>=70].groupby(["school_name"])["reading_score"].count()
number_pass_reading = pd.DataFrame(number_pass_reading)
number_pass_reading = number_pass_reading.rename(columns={"reading_score":"Reading Score"})
percent_pass_reading = (number_pass_reading["Reading Score"]/Total_students["Total Students"])*100
percent_pass_reading = pd.DataFrame(percent_pass_reading)
percent_pass_reading = percent_pass_reading.rename(columns = {0:"% Passing Reading"})

#◾% Overall Passing (The percentage of students that passed math and reading.) Este es imporante por que es la conjunción de ambos condicionales
number_pass_both = merge.loc[pass_students_math & pass_students_reading].groupby(["school_name"])["Student ID"].count()
number_pass_both = pd.DataFrame(number_pass_both)
number_pass_both = number_pass_both.rename(columns={"Student ID": "Total_Both"})
overall_pass = (number_pass_both["Total_Both"]/Total_students["Total Students"])*100
overall_pass = pd.DataFrame(overall_pass)
overall_pass = overall_pass.rename(columns = {0:"% Overall Passing"})

In [663]:
# Creating a data frame to summary all the previous information
A = pd.merge(School_Type, Total_students, left_index=True, right_index=True)
B = pd.merge(A, Total_budget_school, left_index=True, right_index=True)
C = pd.merge(B, Budget_pstudent, left_index=True, right_index=True)
D = pd.merge(C, Average_math_school, left_index=True, right_index=True)
E = pd.merge(D, Average_reading_school, left_index=True, right_index=True)
F = pd.merge(E, percent_pass_math, left_index=True, right_index=True)
G = pd.merge(F, percent_pass_reading, left_index=True, right_index=True)
# School summary
H = pd.merge(G, overall_pass, left_index=True, right_index=True)
H

type  Total Students  Total Budget School  \
school_name                                                            
Huang High School      District            2917              1910635   
Figueroa High School   District            2949              1884411   
Shelton High School     Charter            1761              1056600   
Hernandez High School  District            4635              3022020   
Griffin High School     Charter            1468               917500   
Wilson High School      Charter            2283              1319574   
Cabrera High School     Charter            1858              1081356   
Bailey High School     District            4976              3124928   
Holden High School      Charter             427               248087   
Pena High School        Charter             962               585858   
Wright High School      Charter            1800              1049400   
Rodriguez High School  District            3999              2547363   
Johnson High School    District            4761              3094650   
Ford High School       District            2739              1763916   
Thomas High School      Charter            1635              1043130   

                       Per Student Budget  Average Math Score  \
school_name                                                     
Huang High School                  655.00               76.63   
Figueroa High School               639.00               76.71   
Shelton High School                600.00               83.36   
Hernandez High School              652.00               77.29   
Griffin High School                625.00               83.35   
Wilson High School                 578.00               83.27   
Cabrera High School                582.00               83.06   
Bailey High School                 628.00               77.05   
Holden High School                 581.00               83.80   
Pena High School                   609.00               83.84   
Wright High School                 583.00               83.68   
Rodriguez High School              637.00               76.84   
Johnson High School                650.00               77.07   
Ford High School                   644.00               77.10   
Thomas High School                 638.00               83.42   

                       Average Reading Score  % Passing Math  \
school_name                                                    
Huang High School                      81.18           65.68   
Figueroa High School                   81.16           65.99   
Shelton High School                    83.73           93.87   
Hernandez High School                  80.93           66.75   
Griffin High School                    83.82           93.39   
Wilson High School                     83.99           93.87   
Cabrera High School                    83.98           94.13   
Bailey High School                     81.03           66.68   
Holden High School                     83.81           92.51   
Pena High School                       84.04           94.59   
Wright High School                     83.95           93.33   
Rodriguez High School                  80.74           66.37   
Johnson High School                    80.97           66.06   
Ford High School                       80.75           68.31   
Thomas High School                     83.85           93.27   

                       % Passing Reading  % Overall Passing  
school_name                                                  
Huang High School                  81.32              53.51  
Figueroa High School               80.74              53.20  
Shelton High School                95.85              89.89  
Hernandez High School              80.86              53.53  
Griffin High School                97.14              90.60  
Wilson High School                 96.54              90.58  
Cabrera High School                97.04              91.33  
Bailey High School                 81.93         

## Top Performing Schools (By % Overall Passing)

* Sort and display the top five performing schools by % overall passing.

In [664]:
School_Summary_Sorted= H.sort_values(by="% Overall Passing", ascending=False)
School_Summary_Sorted.head(5)

type  Total Students  Total Budget School  \
school_name                                                         
Cabrera High School  Charter            1858              1081356   
Thomas High School   Charter            1635              1043130   
Griffin High School  Charter            1468               917500   
Wilson High School   Charter            2283              1319574   
Pena High School     Charter             962               585858   

                     Per Student Budget  Average Math Score  \
school_name                                                   
Cabrera High School              582.00               83.06   
Thomas High School               638.00               83.42   
Griffin High School              625.00               83.35   
Wilson High School               578.00               83.27   
Pena High School                 609.00               83.84   

                     Average Reading Score  % Passing Math  % Passing Reading  \
school_name                                                                     
Cabrera High School                  83.98           94.13              97.04   
Thomas High School                   83.85           93.27              97.31   
Griffin High School                  83.82           93.39              97.14   
Wilson High School                   83.99           93.87              96.54   
Pena High School                     84.04           94.59              95.95   

                     % Overall Passing  
school_name                             
Cabrera High School              91.33  
Thomas High School               90.95  
Griffin High School              90.60  
Wilson High School               90.58  
Pena High School                 90.54

# <span style="color:green">Performing by school: looking for trends</span>

#### Sorting the five worst and the five best schools by % Overall Passing we can see that there is two types of schools: 

* Charter.- For each enrolled homeschooler, the public school district receives state and federal public tax money.chool that receives government funding but operates independently of the established state school system in which it is located. As public schools, charter schools are open to all children, do not require entrance exams, cannot charge tuition, and must participate in state testing and federal accountability program.


* District.- These schools must adhere to education standards set by the state education board and are not exempt from any state, federal, or local laws regarding education. They are governed by the school district, which is run by a democratically-elected school board 

##### Having this in mind the top five schools based on % overall passing, we found that all schools are charter type and the worse ranked you can find the district schools. There is not a clear explanation about it, but you can find that district schools have also a greater number of students (they are bigger) than charter schools. Perhaps this big size is a driver about how much attention the scholar institution can provide to every student.

##### This conclusion is reinforced by the idea that the average budget allocated to each student is almost the same, so there is not a budget issue, it shoul be explained by other factors. The best school (Cabrera High School) even has the minimum budget per student and the best performance, and has half of the students in a disctrict school.

## Bottom Performing Schools (By % Overall Passing)

* Sort and display the five worst-performing schools by % overall passing.

In [665]:
School_Summary_Sorted= H.sort_values(by="% Overall Passing", ascending=True)
School_Summary_Sorted.head(5)

type  Total Students  Total Budget School  \
school_name                                                            
Rodriguez High School  District            3999              2547363   
Figueroa High School   District            2949              1884411   
Huang High School      District            2917              1910635   
Hernandez High School  District            4635              3022020   
Johnson High School    District            4761              3094650   

                       Per Student Budget  Average Math Score  \
school_name                                                     
Rodriguez High School              637.00               76.84   
Figueroa High School               639.00               76.71   
Huang High School                  655.00               76.63   
Hernandez High School              652.00               77.29   
Johnson High School                650.00               77.07   

                       Average Reading Score  % Passing Math  \
school_name                                                    
Rodriguez High School                  80.74           66.37   
Figueroa High School                   81.16           65.99   
Huang High School                      81.18           65.68   
Hernandez High School                  80.93           66.75   
Johnson High School                    80.97           66.06   

                       % Passing Reading  % Overall Passing  
school_name                                                  
Rodriguez High School              80.22              52.99  
Figueroa High School               80.74              53.20  
Huang High School                  81.32              53.51  
Hernandez High School              80.86              53.53  
Johnson High School                81.22              53.54

## Math Scores by Grade

* Create a table that lists the average Reading Score for students of each grade level (9th, 10th, 11th, 12th) at each school.

  * Create a pandas series for each grade. Hint: use a conditional statement.
  
  * Group each series by school
  
  * Combine the series into a dataframe
  
  * Optional: give the displayed data cleaner formatting

In [666]:
students_fw["grade"].unique()

array(['9th', '12th', '11th', '10th'], dtype=object)

In [667]:
students_fw["grade"].count()

39170

In [668]:
nine = pd.DataFrame(students_fw.loc[students_fw["grade"]=="9th"].groupby(["school_name"])["math_score"].mean())
nine = nine.rename(columns = {"math_score":"9th"})
ten = pd.DataFrame(students_fw.loc[students_fw["grade"]=="10th"].groupby(["school_name"])["math_score"].mean())
ten = ten.rename(columns = {"math_score":"10th"})
eleven = pd.DataFrame(students_fw.loc[students_fw["grade"]=="11th"].groupby(["school_name"])["math_score"].mean())
eleven = eleven.rename(columns = {"math_score":"11th"})
twelve = pd.DataFrame(students_fw.loc[students_fw["grade"]=="12th"].groupby(["school_name"])["math_score"].mean())
twelve = twelve.rename(columns = {"math_score":"12th"})

m1 = nine.merge(ten, left_index=True, right_index=True)
m2 = m1.merge(eleven, left_index=True, right_index=True)
m3 = m2.merge(twelve, left_index=True, right_index=True)
m3

9th  10th  11th  12th
school_name                                  
Bailey High School    77.08 77.00 77.52 76.49
Cabrera High School   83.09 83.15 82.77 83.28
Figueroa High School  76.40 76.54 76.88 77.15
Ford High School      77.36 77.67 76.92 76.18
Griffin High School   82.04 84.23 83.84 83.36
Hernandez High School 77.44 77.34 77.14 77.19
Holden High School    83.79 83.43 85.00 82.86
Huang High School     77.03 75.91 76.45 77.23
Johnson High School   77.19 76.69 77.49 76.86
Pena High School      83.63 83.37 84.33 84.12
Rodriguez High School 76.86 76.61 76.40 77.69
Shelton High School   83.42 82.92 83.38 83.78
Thomas High School    83.59 83.09 83.50 83.50
Wilson High School    83.09 83.72 83.20 83.04
Wright High School    83.26 84.01 83.84 83.64

## Reading Score by Grade 

* Perform the same operations as above for reading scores

In [669]:
niner = pd.DataFrame(students_fw.loc[students_fw["grade"]=="9th"].groupby(["school_name"])["reading_score"].mean())
niner = niner.rename(columns = {"reading_score":"9th"})
tenr = pd.DataFrame(students_fw.loc[students_fw["grade"]=="10th"].groupby(["school_name"])["reading_score"].mean())
tenr = tenr.rename(columns = {"reading_score":"10th"})
elevenr = pd.DataFrame(students_fw.loc[students_fw["grade"]=="11th"].groupby(["school_name"])["reading_score"].mean())
elevenr = elevenr.rename(columns = {"reading_score":"11th"})
twelver = pd.DataFrame(students_fw.loc[students_fw["grade"]=="12th"].groupby(["school_name"])["reading_score"].mean())
twelver = twelver.rename(columns = {"reading_score":"12th"})

m1r = niner.merge(tenr, left_index=True, right_index=True)
m2r = m1r.merge(elevenr, left_index=True, right_index=True)
m3r = m2r.merge(twelver, left_index=True, right_index=True)
m3r

9th  10th  11th  12th
school_name                                  
Bailey High School    81.30 80.91 80.95 80.91
Cabrera High School   83.68 84.25 83.79 84.29
Figueroa High School  81.20 81.41 80.64 81.38
Ford High School      80.63 81.26 80.40 80.66
Griffin High School   83.37 83.71 84.29 84.01
Hernandez High School 80.87 80.66 81.40 80.86
Holden High School    83.68 83.32 83.82 84.70
Huang High School     81.29 81.51 81.42 80.31
Johnson High School   81.26 80.77 80.62 81.23
Pena High School      83.81 83.61 84.34 84.59
Rodriguez High School 80.99 80.63 80.86 80.38
Shelton High School   84.12 83.44 84.37 82.78
Thomas High School    83.73 84.25 83.59 83.83
Wilson High School    83.94 84.02 83.76 84.32
Wright High School    83.83 83.81 84.16 84.07

## Scores by School Spending

* Create a table that breaks down school performances based on average Spending Ranges (Per Student). Use 4 reasonable bins to group school spending. Include in the table each of the following:
  * Average Math Score
  * Average Reading Score
  * % Passing Math
  * % Passing Reading
  * Overall Passing Rate (Average of the above two)

In [670]:
#creating beans and names
bins = [0,584,629,644,675]
names = ["<$584", "$585-629", "630-644","645-675"]
#only to be sure about beans and distribution
dist = H["Per Student Budget"].mean()
H.dtypes

type                      object
Total Students             int64
Total Budget School        int64
Per Student Budget       float64
Average Math Score       float64
Average Reading Score    float64
% Passing Math           float64
% Passing Reading        float64
% Overall Passing        float64
dtype: object

In [671]:
I = H
I["Spending Range(Per Student)"]=pd.cut(I["Per Student Budget"], bins, labels=names)
I = I.groupby("Spending Range(Per Student)")
I = I[["Average Math Score", "Average Reading Score", "% Passing Math", "% Passing Reading", "% Overall Passing"]]
pd.options.display.float_format = '{:,.2f}'.format
I.mean()

Average Math Score  Average Reading Score  \
Spending Range(Per Student)                                              
<$584                                     83.46                  83.93   
$585-629                                  81.90                  83.16   
630-644                                   78.52                  81.62   
645-675                                   77.00                  81.03   

                             % Passing Math  % Passing Reading  \
Spending Range(Per Student)                                      
<$584                                 93.46              96.61   
$585-629                              87.13              92.72   
630-644                               73.48              84.39   
645-675                               66.16              81.13   

                             % Overall Passing  
Spending Range(Per Student)                     
<$584                                    90.37  
$585-629                                 81.42  
630-644                                  62.86  
645-675                                  53.53

#  <span style="color:green">Looking for more trends</span>

* As we saw previously, the explanation regarding school performance is not related with money (budget per student)
* In this table we can confirm that the spending range less than $584 is even better, based in score and overall passing, than more expensive schools.
* Definitely the performance is more explained by how charter schools are doing his job at least regarding with student. 

#### I'd like to remark that this analysis requieres more time anda data, but at first glance we can conclude those aspects.

* As we saw previously, the explanation regarding school performance is not related with money (budget per student)
* In this table we can confirm that the spending range less than $584 is even better, based in score and overall passing, than more expensive schools.
* Definitely the performance is more explained by how charter schools are doing his job at least regarding with student. 

#### I'd like to remark that this analysis requieres more time anda data, but at first glance we can conclude those aspects.

## Scores by School Size

* Perform the same operations as above, based on school size.

In [672]:
J=H
bins2 = [0,999,1999,5000]
names2 = ["Small (<1000)", "Medium (1000-2000)", "Large (2000-5000)"]
J.dtypes

type                             object
Total Students                    int64
Total Budget School               int64
Per Student Budget              float64
Average Math Score              float64
Average Reading Score           float64
% Passing Math                  float64
% Passing Reading               float64
% Overall Passing               float64
Spending Range(Per Student)    category
dtype: object

In [673]:
J["School Size"]= pd.cut(J["Total Students"], bins2, labels=names2, include_lowest=True)
J = J.groupby("School Size")
J = J[["Average Math Score", "Average Reading Score", "% Passing Math", "% Passing Reading", "% Overall Passing"]]
pd.options.display.float_format = '{:,.2f}'.format
J.mean()

Average Math Score  Average Reading Score  % Passing Math  \
School Size                                                                     
Small (<1000)                    83.82                  83.93           93.55   
Medium (1000-2000)               83.37                  83.86           93.60   
Large (2000-5000)                77.75                  81.34           69.96   

                    % Passing Reading  % Overall Passing  
School Size                                               
Small (<1000)                   96.10              89.88  
Medium (1000-2000)              96.79              90.62  
Large (2000-5000)               82.77              58.29

# <span style="color:green">Looking for more trends</span> 

* The school size and type are definetively drivers as we can see in the last tables. 



## Scores by School Type

* Perform the same operations as above, based on school type

In [680]:
K = H.groupby("type")
K = K[["Average Math Score", "Average Reading Score", "% Passing Math", "% Passing Reading", "% Overall Passing"]]
K.mean()

Average Math Score  Average Reading Score  % Passing Math  \
type                                                                  
Charter                83.47                  83.90           93.62   
District               76.96                  80.97           66.55   

          % Passing Reading  % Overall Passing  
type                                            
Charter               96.59              90.43  
District              80.80              53.67

In [681]:
pwd

'd:\\Users\\rodrigo.guarneros.IFT\\Desktop\\HW1\\PyCitySchools'